scale/scheduler/models.py

```diff
@@ -60,6 +69,12 @@ def update_scheduler(self, new_data):
         :type new_data: dict
         """
 
+        if 'resource_level' in new_data:
+            try:
+                validate_resource_level(new_data['resource_level'])
+            except ValidationError:
+                logger.exception('Invalid resource level given.')
+                raise 
         sched = self.select_for_update().filter(id=1)
         sched.update(**new_data)
```

Comment:
The `@transaction.atomic` and `select_for_update()` are unnecessary (this is very old code btw). A simple update call is sufficient (not even necessary to filter really since there is just one row).